docs/rfcs/0002-session-id-sharing.md

```diff
@@ -0,0 +1,226 @@
+
+There will be four key triggers that will handle the shared_session_id: initialization, on VPN activation, ping timer and on VPN deactivation.
+
+### ON VPN ACTIVATION
+
+#### MAIN APPLICATION
+
+In case the VPN is started from the main application, a session id will be created and shared through the activation signal with the daemon. The generated UUID will be stored in the main application local storage and the session ping will be submitted.
+
+#### DAEMON
+
+If the VPN is being activated through the main application, a session id will have been passed through the activation signal. Otherwise the Daemon will generate this value itself. The UUID will be stored in the daemon local storage and the session ping will be submitted.
+
+### ON PING TIMER
+
+#### MAIN APPLICATION
+
+The session id is retrieved from local storage and recorded. The session ping is submitted.
+
+#### DAEMON
+
+The session id is retrieved from local storage and recorded. The session ping is submitted.
+
+### ON VPN DEACTIVATION
+
+The deactivation signal may not be hit in both the main application and the daemon, because both processes may be force killed. The local storage that contains the session id will be cleared after submitting the final session ping i.e. the ping sent upon VPN deactivation. This will be a flag at initialization that the final session ping was not submitted and there may be lingering data from a previous session in storage that needs to be submitted.
+
+#### MAIN APPLICATION
+
+The session id is retrieved from local storage and recorded. The session ping is submitted. Finally local storage for the session id is cleared.
+
+#### DAEMON
+
+The session id is retrieved from local storage and recorded. The session ping is submitted. Finally local storage for the session id is cleared.
+
+### ON INITIALIZATION
+
+#### MAIN APPLICATION
+
+The session id is retrieved from local storage. If it is not empty, there are four possibilities:
+
+1. The same session from a previous app run is still ongoing. No action is required
+2. Another session is ongoing in which case a flush ping is required to submit data from a previous session and start recording data for the new session
+3. No session is ongoing but there is a lingering session id indicating that there is still lingering data from a previous session in storage and a flush ping is required
+4. No session is ongoing and no lingering session id is in storage. No action is required
+
+#### DAEMON
+
+On the daemon there are only two possibilities:
+
+1. No session is ongoing but there is a lingering session id indicating that there is still lingering data from a previous session in storage and a flush ping is required
+2. No session is ongoing and no lingering session id is in storage. No action is required
+
+## APPENDIX
+
+### APPENDIX 1: PSEUDO CODE
+
+#### Main application
+
+```cpp
+MainApp::init() {
```

Comment:
What if it the app local storage is empty but there is an active VPN session in the daemon? I think we're missing that case.

Reply:
Good catch, indeed that is missing.

Reply:
Let me know when that case is added, and I'll re-review. Thanks.

Reply:
The rewrite makes this obsolete. Let me know if you agree, for now resolving.